packer/payload/prepare-ami.sh

```diff
@@ -0,0 +1,95 @@
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+SOURCE_DIR="$(cd "$(dirname "$0")"; pwd)"
+KUBERNETES_RELEASE="v1.5.4"
+KUBEADM_RELEASE="v1.6.0-alpha.0.2074+a092d8e0f95f52"
+CNI_RELEASE="07a8a28637e97b22eb8dfe710eeae1344f69d16e"
+
+## Make sure we get the latest updates since the base image was released
+apt-get update -q
+apt-get upgrade -qy
+
+## TODO: Update docker to use overlay2 by default
+apt-get install -qy docker.io jq python-pip python-setuptools ebtables socat ntp
+
+## Install official Kubernetes binaries
+mkdir /tmp/kubebin
+(
+  cd /tmp/kubebin
+  curl -sf -O "https://storage.googleapis.com/kubernetes-release/release/${KUBERNETES_RELEASE}/bin/linux/amd64/kubelet"
+  curl -sf -O "https://storage.googleapis.com/kubernetes-release/release/${KUBERNETES_RELEASE}/bin/linux/amd64/kubectl"
+  curl -sf -O "https://storage.googleapis.com/kubernetes-release-dev/ci-cross/${KUBEADM_RELEASE}/bin/linux/amd64/kubeadm"
+  curl -sf -O "https://storage.googleapis.com/kubernetes-release/network-plugins/cni-amd64-${CNI_RELEASE}.tar.gz"
+
+  install -o root -g root -m 0755 ./kubeadm /usr/bin/kubeadm
+  install -o root -g root -m 0755 ./kubectl /usr/bin/kubectl
+  install -o root -g root -m 0755 ./kubelet /usr/bin/kubelet
+
+  # Also install CNI
+  mkdir /opt/cni
+  (
+    cd /opt/cni
+    tar -xzf "/tmp/kubebin/cni-amd64-${CNI_RELEASE}.tar.gz"
+  )
+)
+rm -rf /tmp/kubebin
+
+## Install systemd scripts for kubelet
+sudo mkdir /etc/systemd/system/kubelet.service.d
+install -o root -g root -m 0600 "${SOURCE_DIR}/systemd/kubelet.service" \
+  /lib/systemd/system/kubelet.service
+install -o root -g root -m 0600 "${SOURCE_DIR}/systemd/10-kubeadm.conf" \
+  /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+install -o root -g root -m 0600 "${SOURCE_DIR}/systemd/20-cloud-provider.conf" \
+  /etc/systemd/system/kubelet.service.d/20-cloud-provider.conf
+systemctl enable kubelet
+
+## We will need AWS tools as well
+pip install "https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-latest.tar.gz"
+pip install awscli
+
+## Pre-fetch various images, so that `kubeadm init` is a bit quicker
+## TODO: pre-fetch add-ons that are layed down by kubeadm.
+## Also, address logging and metrics
+images=(
+  "gcr.io/google_containers/kube-proxy-amd64:${KUBERNETES_RELEASE}"
+  "gcr.io/google_containers/kube-apiserver-amd64:${KUBERNETES_RELEASE}"
+  "gcr.io/google_containers/kube-scheduler-amd64:${KUBERNETES_RELEASE}"
+  "gcr.io/google_containers/kube-controller-manager-amd64:${KUBERNETES_RELEASE}"
+  "gcr.io/google_containers/etcd-amd64:3.0.14-kubeadm"
+  "gcr.io/google_containers/kube-discovery-amd64:1.0"
+  "gcr.io/google_containers/pause-amd64:3.0"
+  "gcr.io/google_containers/etcd:2.2.1"
```

Comment:
I don't see a pull for 3.x.y anywhere?

Does calico only support 2.2.1?  That super old with a bunch of known issues.